src/methods/clipping/union.jl

```diff
@@ -0,0 +1,316 @@
+                # I use ext_poly_b here instead of poly_b in order to not overcount
+                # area that is a hole of poly_a and in a hole of poly_b
+                new_hole = difference(_lin_ring_to_poly(hole), ext_poly_b)
+                for h in new_hole
+                    if length(h)>0
+                        # I claim I can index h at one, because it can't
+                        # have a hole because a hole within a hole would
+                        # be an invalid polygon
+                        append!(mult_poly, [h[1]])
+                    end
+                end        
+            end
+            
+            for hole in GI.gethole(poly_b)
+                new_hole = difference(_lin_ring_to_poly(hole), poly_a)
+                for h in new_hole
+                    if length(h)>0
+                        append!(mult_poly, [h[1]])
+                    end
+                end 
+            end
+
+            push!(final_polys, mult_poly)
+        end
+
+        return final_polys
+    end
+        
+end
+
+
+"""
+    union(
+        ::GI.PolygonTrait, poly_a,
+        ::GI.PolygonTrait, poly_b,
+    )::Vector{Vector{Vector{Tuple{Float64}}}}    
+
+Calculates the union between two polygons.
+## Example
+
+```jldoctest
+import GeoInterface as GI, GeometryOps as GO
+
+p1 = GI.Polygon([[(0.0, 0.0), (5.0, 5.0), (10.0, 0.0), (5.0, -5.0), (0.0, 0.0)]])
+p2 = GI.Polygon([[(3.0, 0.0), (8.0, 5.0), (13.0, 0.0), (8.0, -5.0), (3.0, 0.0)]])
+GO.union(p1, p2)
+
+# output
+1-element Vector{Vector{Vector{Tuple{Float64, Float64}}}}:
+[[[(6.5, 3.5), (5.0, 5.0), (0.0, 0.0), (5.0, -5.0), (6.5, -3.5), (8.0, -5.0), (13.0, 0.0), (8.0, 5.0), (6.5, 3.5)]]]
+```
+"""
+function union(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b)
+    # First, I get the exteriors of the two polygons
+    ext_poly_a = GI.getexterior(poly_a)
+    ext_poly_a = _lin_ring_to_poly(ext_poly_a)
+    ext_poly_b = GI.getexterior(poly_b)
+    ext_poly_b = _lin_ring_to_poly(ext_poly_b)
+    # Then, I get the union of the exteriors
+    a_list, b_list, a_idx_list, intr_list, edges_a, edges_b = _build_ab_list(GI.trait(ext_poly_a), ext_poly_a, GI.trait(ext_poly_b), ext_poly_b)
+    temp = _trace_union(GI.trait(ext_poly_a), ext_poly_a, GI.trait(ext_poly_b), ext_poly_b, a_list, b_list, a_idx_list, intr_list, edges_a, edges_b)
+    polys = temp[1]
+    diff_polys = temp[2]
+    # If the original polygons had holes, we call '_get_union_holes' to take that
+    # into account.
+    if GI.nhole(poly_a)==0 && GI.nhole(poly_b)==0
+        if !diff_polys
+            return [polys]
+        else
+            final_polys =  Vector{Vector{Vector{Tuple{Float64, Float64}}}}(undef, length(polys))
+            for i in 1:length(polys)
+                final_polys[i] = [polys[i]]
+            end
+            return final_polys
+        end
+    else
+        return _get_union_holes(polys, poly_a, poly_b, ext_poly_b, diff_polys)
+    end 
+end
+
+union(geom_a, geom_b) =
```

Comment:
Same commentary as difference. These boys go to the top since they are most important. You only need examples for functions that users are going to use. Remove TODOs and if the issues/questions still exist then open an issue